src/reducers-actions/trades.ts

```diff
@@ -161,28 +186,97 @@ function applyRevertsToTrades(
   return [flattenMapOfLists(tradesByRevertKey), getPendingTrades(tradesByRevertKey)]
 }
 
+// ******** REDUCER
+
 export const reducer = (state: TradesState, action: ReducerActionType): TradesState => {
   switch (action.type) {
     case 'APPEND_TRADES': {
-      const { trades: currTrades, pendingTrades: currPendingTrades } = state
-      const { trades: newTrades, reverts, lastCheckedBlock } = action.payload
+      const { trades: newTrades, reverts, lastCheckedBlock, networkId } = action.payload
+      const { trades: currTrades, pendingTrades: currPendingTrades } = state[networkId]
 
       const [trades, pendingTrades] = applyRevertsToTrades(newTrades, reverts, currPendingTrades)
 
-      return { trades: currTrades.concat(trades), lastCheckedBlock, pendingTrades }
+      return { ...state, [networkId]: { trades: currTrades.concat(trades), lastCheckedBlock, pendingTrades } }
     }
     case 'OVERWRITE_TRADES': {
-      const { trades: newTrades, reverts, lastCheckedBlock } = action.payload
+      const { trades: newTrades, reverts, lastCheckedBlock, networkId } = action.payload
 
       const [trades, pendingTrades] = applyRevertsToTrades(newTrades, reverts)
 
-      return { trades, lastCheckedBlock, pendingTrades }
+      return { ...state, [networkId]: { trades, lastCheckedBlock, pendingTrades } }
     }
     case 'UPDATE_BLOCK': {
-      return { ...state, ...action.payload }
+      const { networkId, lastCheckedBlock } = action.payload
+
+      return { ...state, [networkId]: { ...state[networkId], lastCheckedBlock } }
     }
     default: {
       return state
     }
   }
 }
+
+// TODO: use the one from David once his changes are merged https://github.com/gnosis/dex-react/pull/1091
+function setStorageItem(key: string, data: unknown): void {
+  // localStorage API accepts only strings
+  // TODO: consider switching to localForage API (accepts all types)
+  const formattedData = JSON.stringify(data)
+  return localStorage.setItem(key, formattedData)
+}
+
+// ******** SIDE EFFECT
+
+export async function sideEffect(state: TradesState, action: ReducerActionType): Promise<void> {
+  switch (action.type) {
+    case 'APPEND_TRADES':
+    case 'OVERWRITE_TRADES':
+    case 'UPDATE_BLOCK':
+      setStorageItem(TRADES_LOCAL_STORAGE_KEY, state)
+  }
```

Comment:
consider adding a default here in case

Reply:
Actually, the function returns `void`, so having a default makes no difference.